src/media/js/elements/categories.js

```diff
@@ -0,0 +1,110 @@
+/*
+    Marketplace Category List element.
+
+    <mkt-category-list>
+        A <ul> wrapper.
+
+        createdCallback - creates <mkt-category-item> elements from the
+                          categories provided by categories.js.
+
+    <mkt-category-item>
+        An <li> element.
+
+        createdCallback - creates <a> elements given its [data-mkt-category]
+                          that a JSON object with category `slug` and `name`.
+
+    z.page.loaded and z.page.navigate
+        - Toggle active-ness of <mkt-category-link>s.
+*/
+define('elements/categories',
+    ['categories', 'core/format', 'core/nunjucks', 'core/urls', 'core/z',
+     'document-register-element', 'jquery', 'underscore'],
+    function(categories, format, nunjucks, urls, z, dre, $, _) {
+    'use strict';
+
+    var LINK = 'mkt-category-link';
+    var LINK_ACTIVE = 'mkt-category-link--active';
+
+    var MktCategoryListElement = document.registerElement('mkt-category-list', {
+        prototype: Object.create(HTMLUListElement.prototype, {
+            createdCallback: {
+                value: function() {
+                    var root = this;
+
+                    var helperDiv = document.createElement('div');
+                    var catTemplate =
+                        '<mkt-category-item ' +
```

Comment:
I really need to get that babel stuff merged in. This would be way nicer with string templates.